pkg/logging/db.go

```diff
@@ -0,0 +1,191 @@
+package logging
+
+import (
+	"encoding/json"
+	"log"
+
+	"github.com/jinzhu/gorm"
+
+	"github.com/jmpsec/osctrl/pkg/backend"
+	"github.com/jmpsec/osctrl/pkg/settings"
+	"github.com/jmpsec/osctrl/pkg/types"
+)
+
+const (
+	// DBName as JSON key for configuration
+	DBName string = "db"
+	// DBFile as default file for configuration
+	DBFile string = "config/" + DBName + ".json"
+)
+
+// OsqueryResultData to log result data to database
+type OsqueryResultData struct {
+	gorm.Model
+	UUID        string `gorm:"index"`
+	Environment string
+	Name        string
+	Action      string
+	Epoch       int64
+	Columns     json.RawMessage
+	Counter     int
+}
+
+// OsqueryStatusData to log status data to database
+type OsqueryStatusData struct {
+	gorm.Model
+	UUID        string `gorm:"index"`
+	Environment string
+	Line        string
+	Message     string
+	Version     string
+	Filename    string
+	Severity    string
+}
+
+// OsqueryQueryData to log query data to database
+type OsqueryQueryData struct {
+	gorm.Model
+	UUID        string `gorm:"index"`
+	Environment string
+	Name        string
+	Data        json.RawMessage
+	Status      int
+}
+
+// LoggerDB will be used to log data using a database
+type LoggerDB struct {
+	Database      *gorm.DB
+	Configuration backend.JSONConfigurationDB
+	Enabled       bool
+}
+
+func CreateLoggerDB() (*LoggerDB, error) {
```

Comment:
i need to get more used to the github review interface, so I haven't been able to see yet how this is called, but would it make sense to make a Logging interface and have each of files implement a different kind of logger?

Reply:
Probably that is the way to go